frontend/middleware_validatestatic.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-logr/logr"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+const (
+	wantedResourceProviderNamespace = "Microsoft.RedHatOpenShift"
+	resourceTypeHCPOpenshiftCluster = "HcpOpenShiftClusters"
+)
+
+var rxResourceGroupName = regexp.MustCompile(`^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
```

Comment:
where does this regex come from? Is there documentation we can comment regarding this?

Reply:
There are rules here: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftresources
They allow these but we don't need them since we only use the English alphabet
TitlecaseLetter,
ModifierLetter,
OtherLetter,